pkg/policy/multi.go

```diff
@@ -145,10 +145,11 @@ func processMultiDenyPolicyResult(
 			builder.resourceType = result.ResourceType
 		}
 
-		if result.Resource != nil {
-			builder.addResource(result.Resource.Key())
+		if resource := result.GetResource(); resource != nil {
```

Comment:
I haven't run this and studied the output but would we not need something like this here:

```go
if result.PrimaryResource != nil {
 builder.setPrimaryResource(result.PrimaryResource)
}
```

Or is that only needed once we want to update the output format?

Reply:
Thanks! That was a good catch. The reason that this was working with my example was because the builder already exists and has the right primary resource if you have a corresponding `resources` rule for the deny. When I commented out the `resources` rule, this issue occurred.

Fixed in 7062437